### PR TITLE
impl TryFrom from native to NonZero

### PR DESCRIPTION
**Pull request by heca-project**
You can now do the following:

    ```
    let p: u8 = 5;
    let q: NonZeroU8 = p.try_into()?;
    ```

### CODE REVIEW

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -110,6 +111,20 @@ assert_eq!(size_of::<Option<core::num::", stringify!($Ty), ">>(), size_of::<", s
...
+            #[stable(feature = "try_from", since = "1.34.0")]
+            impl TryFrom<$Int> for $Ty {
+               type Error = TryFromIntError;
+               fn try_from(n: $Int) -> Result<Self, Self::Error> {
+                   if n != 0 {
+                        // SAFETY: we just checked that there's no `0`
+                        Ok(unsafe { Self(n) })
+                   } else {
+                        Err(TryFromIntError(()))
+                   }
```
Comment (kennytm) [inline src/libcore/num/mod.rs:124]:
```suggestion
                   Self::new(n).ok_or(TryFromIntError(()))
```

**Comment by kennytm** [comment]
still a 

```
tidy error: /checkout/src/libcore/num/mod.rs:114: trailing whitespace
```

----

r? @rust-lang/libs

**Comment by poliorcetics** [comment]
In #72717 I had tests written for this if you want them. Writing a `doc_comment!(...)` like the one a few lines above might nice too.

**Review by dtolnay — COMMENTED** [review]
I'll close this in favor of #72717 which was opened first and has the same impls along with tests.

Thanks anyway!

**Pull request closed without merging**